bugbug/tools/code_review.py

```diff
@@ -516,6 +624,191 @@ def bug_title(self) -> str:
     def patch_title(self) -> str:
         return self._revision_metadata["fields"]["title"]
 
+    @property
+    def revision_id(self) -> int:
+        return self._revision_metadata["id"]
+
+    def _get_transactions(self) -> list[dict]:
+        assert phabricator.PHABRICATOR_API is not None
+
+        transactions = phabricator.PHABRICATOR_API.request(
+            "transaction.search",
+            objectIdentifier=self._revision_metadata["phid"],
+        )["data"]
+
+        return transactions
+
+    def get_comments(
+        self,
+    ) -> Iterable[PhabricatorInlineComment | PhabricatorGeneralComment]:
+        transactions = self._get_transactions()
+
+        for transaction in transactions:
+            comment = phabricator_transaction_to_comment(transaction)
+            if comment:
+                yield comment
+
+    def to_md(self) -> str:
+        """Generate a comprehensive, LLM-friendly markdown representation of the patch.
+
+        Returns a well-structured markdown document that includes revision metadata,
+        diff information, stack information, code changes, and comments.
+        """
+        # TODO: print authors' names instead of PHIDs
+
+        date_format = "%Y-%m-%d %H:%M:%S"
+        md_lines = []
+
+        revision = self._revision_metadata
+        md_lines.append(f"# Revision D{revision['id']}: {revision['fields']['title']}")
+        md_lines.append("")
+
+        md_lines.append("## Basic Information")
+        md_lines.append(f"- **URI**: {revision['fields']['uri']}")
+        md_lines.append(f"- **Revision Author**: {revision['fields']['authorPHID']}")
+        md_lines.append(f"- **Title**: {revision['fields']['title']}")
+        md_lines.append(f"- **Status**: {revision['fields']['status']['name']}")
+        md_lines.append(f"- **Created**: {self.date_created.strftime(date_format)}")
+        md_lines.append(f"- **Modified**: {self.date_modified.strftime(date_format)}")
+
+        bug_id = revision["fields"].get("bugzilla.bug-id") or "N/A"
+        md_lines.append(f"- **Bugzilla Bug**: {bug_id}")
+
+        summary = revision["fields"].get("summary")
+        if summary:
+            md_lines.append("## Summary")
+            md_lines.append("")
+            md_lines.append(summary)
+            md_lines.append("")
+
+        test_plan = revision["fields"].get("testPlan")
+        if test_plan:
+            md_lines.append("## Test Plan")
+            md_lines.append("")
+            md_lines.append(test_plan)
+            md_lines.append("")
+
+        md_lines.append("")
+
+        md_lines.append("## Diff Information")
```

Comment:
Maybe it would be useful to know how many diffs (is this the first diff, or has it been updated X times?). Not sure

Reply:
We don't have it out of the box; we need to query all diffs instead of just the last one. I will have it as a follow-up.